src/Connection/Http2Connection.php

```diff
@@ -1444,8 +1449,13 @@ private function shutdown(?int $lastId = null, ?\Throwable $reason = null): Prom
         $this->socket->close();
 
         if (!empty($this->streams)) {
-            $exception = new SocketException("Connection closed", $reason->getCode(), $reason);
+            $reason = $exception = $reason ?? new SocketException("Connection closed");
             foreach ($this->streams as $id => $stream) {
+                $exception = $reason;
+                if ($id > $lastId && $stream->request !== null) {
```

Comment:
I'm not sure why `$lastId` is defined the way it is. It seems like in `$lastId = $lastId ?? ($id ?? 0);` `$id` is always undefined and why does it fall back to `0`? While it may make sense for the written `GOAWAY` frame, these requests shouldn't count as unprocessed.

Reply:
Copy/paste failure from `http-server`. `$id` there is set by a foreach loop that I removed. Will fix.